### PR TITLE
Fix: do not allow resizing card to negative values

### LINKED ISSUE #31: [BUG] Sidebar displays negative values after resizing the image
When resizing image, it's allowed to resize it down to negative value (according to the sidebar). When resizing the image back, sidebar says it's still a negative value, despite image being resized normally.
image

### PR DESCRIPTION
**Pull request by tryonelove**
Closes #31 

Speaking of doing that in actions, it won't work. Since we're calling `callback` inside actions, we don't know if size is going increase or decrease, so we'll get negative values.

### CODE REVIEW

**Comment by subhoghoshX** [comment]
In the actions, I'm not setting the increase of decrease, I'm setting the absolute values there. And your implementation still allows the properies to be negative from the side panel.

**Comment by subhoghoshX** [comment]
In the actions, I'm not setting the increase or decrease, I'm setting the absolute values there. And your implementation still allows the properties to be negative from the side panel.

**Comment by jalezi** [comment]
> In the actions, I'm not setting the increase or decrease, I'm setting the absolute values there. And your implementation still allows the properties to be negative from the side panel.

@subhoghoshX I was just about to ask if side panel negative values on user input should implemented in this PR and if `<input type="text`/> is intentional.
Also opacity should not be greater than 100. What about radius? Should we set max radius?

**Comment by jalezi** [comment]
> In the actions, I'm not setting the increase or decrease, I'm setting the absolute values there. And your implementation still allows the properties to be negative from the side panel. 

I can fix 2nd part negative properties in side panel. If you agree @subhoghoshX  & @tryonelove. No pressure.

I have plenty of time ⏳

**Comment by tryonelove** [comment]
@subhoghoshX
FYI introducing new solution. For now, it's not allowed to enter or resize to negative values. Also applied the fix to radius and opacity.

**Comment by subhoghoshX** [comment]
> @subhoghoshX FYI introducing new solution. For now, it's not allowed to enter or resize to negative values. Also applied the fix to radius and opacity.

Looking almost good. First, revert the changes you did in `CardResizable.tsx` as that's not necessary, and also in the current implementation, you'll notice some flickering when resizing to a negative and then positive again.

I've pushed a better approach which is just updating the store through a buffer, rather than directly.

**Comment by tryonelove** [comment]
@subhoghoshX pushed more fixes.

File: components/TweetCard/CardResizable.tsx [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import { useCardStore } from "../../store/card";
-import { Resizable } from "re-resizable";
+import { Resizable, ResizeDirection } from "re-resizable";
```
Comment (subhoghoshX) [inline components/TweetCard/CardResizable.tsx:2]:
```suggestion
import { Resizable } from "re-resizable";
```

File: components/TweetCard/CardResizable.tsx [hunk 2]
```diff
@@ -56,6 +68,7 @@ export default function ResizableTweet({ rootRef }: any) {
     <Resizable
       size={{ width: cardWidth, height: cardHeight }}
       onResize={resizeHandler}
+      onResizeStop={resizeStopHandler}
       className="lg:-translate-x-24"
       minHeight={300}
       minWidth={500}
```
Comment (subhoghoshX) [inline components/TweetCard/CardResizable.tsx:74]:
```suggestion
      minHeight={MIN_ALLOWED_HEIGHT}
      minWidth={MIN_ALLOWED_WIDTH}
```

**Review by subhoghoshX — APPROVED** [review]

**Comment by subhoghoshX** [comment]
Thank you :)

**Pull request merged**